web_src/js/index.js

```diff
@@ -2167,6 +2167,81 @@ function searchRepositories() {
   });
 }
 
+function hitAllKeys(keys, entry) {
+  let i = 0;
+  let j = 0;
+  const hitIndexes = [];
+  while (i < keys.length) {
+    for (; j < entry.length; j++) {
+      if (keys[i] === entry[j]) {
+        hitIndexes.push(j);
+        j++;
+        break;
+      }
+    }
+    i++;
+  }
+
+  return hitIndexes;
+}
+
+function addHighLightToHit($a, entry, indexes) {
+  let highLightText = '';
+  for (let i = 0; i < entry.length; i++) {
+    if (indexes.includes(i)) {
+      highLightText += `<b>${entry[i]}</b>`;
+    } else {
+      highLightText += entry[i];
+    }
+  }
+  $a.html(highLightText);
```

Comment:
Can you use https://api.jquery.com/wrap/ instead of `.html` (which is generally better to be avoided where possible for security reasons).

Reply:
Hi silverwind, 

Thanks for your suggestion. But, I found .Wrap() is different to .html(), which is to wrap element as same as .appendTo(), not going to convert string to html.

Also, the strings in html is from my own code, not from untrusted sources. So, there is no any security reasons for .html.

And there are many .html used in web_src/js/index.js...

Reply:
Yeah, it does not matter in this case as the wrapped HTML is already rendered before by the server. I just generally prefer to reduce `.html` usage.

Reply:
A range-based escaping algorithm should be used. For example:

```
keys = "the/directory"
entry = "thedir"
```

Then the result should be:  

```
<span class=red>${escape("the")}</span>
<span>${escape("/")}
<span class=red>${escape("dir")}</span>
<span>${escape("ectory")}</span>
```

Otherwise, if the filename contains `<br>` or some JS code, the viewer may be attacked by XSS.